snuba/admin/static/runtime_config/auditlog.tsx

```diff
@@ -0,0 +1,104 @@
+import React, { ReactNode, useState } from "react";
+
+import Client from "../api_client";
+import { containerStyle, paragraphStyle } from "./styles";
+import { Table } from "../table";
+import { ConfigValue } from "./types";
+
+function AuditLog(props: { api: Client }) {
+  const { api } = props;
+  const [data, setData] = useState<any[] | null>(null);
+
+  function fetchData() {
+    api.getAuditlog().then((res) => {
+      setData(res);
+    });
+  }
+
+  if (data === null) {
+    fetchData();
+  }
+
+  if (!data) {
+    return null;
+  }
+
+  const rowData = data.map(({ key, timestamp, user, before, after }) => {
+    const formattedDate = new Date(timestamp * 1000).toLocaleDateString(
+      "en-US",
+      {
+        day: "numeric",
+        month: "short",
+        year: "numeric",
+        hour: "numeric",
+        minute: "numeric",
+        second: "numeric",
+        timeZoneName: "short",
+      }
+    );
+
+    return [
+      formattedDate,
+      user,
+      <code>{key}</code>,
+      getActionDetail(before, after),
+    ];
+  });
+
+  const usersTimeZone = Intl.DateTimeFormat().resolvedOptions().timeZone;
+
+  return (
+    <div style={containerStyle}>
+      <p style={paragraphStyle}>
+        Only changes to runtime configuration are currently being captured in
+        the audit log
+      </p>
+      <Table
+        headerData={[`Timestamp (${usersTimeZone})`, "User", "Key", "Action"]}
+        rowData={rowData}
+        columnWidths={[1, 1, 1, 1]}
+      ></Table>
+    </div>
+  );
+}
+
+function getActionDetail(
+  before: ConfigValue | null,
```

Comment:
My comment below about optionals applies here

Reply:
If we made them optional instead of nullable, then I don't think we could pass an after to this function without a before, as that comes later in the list of arguments.